### PR TITLE
fix(TDKN-190): serialize the version number when serializing properties

### PR DESCRIPTION
**Pull request by akhabali**
**What is the problem this Pull Request is trying to solve?**
 https://jira.talendforge.org/browse/TDKN-190

**What is the chosen solution to this problem?**
serialize property version with the properties
 
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/33/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/33/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/34/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/34/)

File: daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java [hunk 1]
```diff
@@ -242,7 +243,7 @@ public void setupLayout() {
     public String toSerialized() {
         handleAllPropertyEncryption(ENCRYPT);
         try {
-            return JsonWriter.objectToJson(this);
+            return SerializerDeserializer.toSerialized(this, false);
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java]: I think you should be using true for the PERSISTANCE flag, cause the serialization here is meant to be persisted in the studio filesystem.

Comment (akhabali) [reply daikon/src/main/java/org/talend/daikon/properties/PropertiesImpl.java]: the flag is never used in the called methods.

**Comment by akhabali** [comment]
@sgandon i have added the test case in the class below. https://github.com/Talend/daikon/blob/c388101afc1ab70be037348ba74967b20b0e93f4/daikon/src/test/java/org/talend/daikon/serialize/jsonio/PersistenceTestObject.java

the test already exist in https://github.com/Talend/daikon/blob/c388101afc1ab70be037348ba74967b20b0e93f4/daikon/src/test/java/org/talend/daikon/serialize/jsonio/SerializeDeserializeTest.java

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/37/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/37/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/44/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/44/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/49/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/49/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/55/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/55/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/67/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/67/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/72/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/72/)

**Pull request merged**